src/mappers/provinces/province_mapper_importer.cpp

```diff
@@ -44,6 +46,42 @@ void VerifyProvinceIsMapped(const mappers::Hoi4ToVic3ProvinceMapping& hoi4_to_vi
    }
 }
 
+std::vector<std::string> AreVic3ProvincesFromSameState(const std::vector<std::string>& provinces_from_map,
+    const std::map<std::string, vic3::StateRegion>& state_regions_from_save)
+{
+   std::map<std::string, std::string> province_to_state_map;
+   for (const auto& state_from_save: state_regions_from_save)
+   {
+      for (const std::string& province_from_map: provinces_from_map)
+      {
+         std::set<std::string> provinces_from_save = state_from_save.second.GetProvinces();
+         if (provinces_from_save.find(province_from_map) != provinces_from_save.end())
+         {
+            province_to_state_map[province_from_map] = state_from_save.first;
+         }
+      }
+   }
+
+   std::set<std::string> state_names;
+   for (const auto& entry: province_to_state_map)
+   {
+      state_names.insert(entry.second);
+   }
+
+   std::vector<std::string> warnings;
+   if (state_names.size() > 1)
+   {
+      for (const auto& province_from_map: provinces_from_map)
+      {
+         std::string warning = "Province " + province_from_map + " is designated as part of " +
```

Comment:
For this check, the problem is that the link has provinces from different states.

Reply:
Perhaps `Province {} is designated as part of {} in Vic3 data and is placed in a mapping with provinces from other states.`

Reply:
fix with [Update province_mapper_importer.cpp](https://github.com/ParadoxGameConverters/Vic3ToHoI4/pull/632/commits/3c8de4b72756fe36ac2632bc8e272ef467c287c0)